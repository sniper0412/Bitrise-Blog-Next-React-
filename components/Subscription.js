import React from 'react';
import cx from 'classnames';
import SVG from 'react-inlinesvg';

export default class Subscription extends React.Component {
  state = {
    email: '',
    success: null,
    error: null,
    isLoading: false
  };

  subscribeUser = async evt => {
    evt.preventDefault();
    this.setState({ isLoading: true });

    const { email } = this.state;

    if (!email) {
      return this.setState({
        success: null,
        error: 'Please provide an email address',
        isLoading: false
      });
    }

    const { success, message } = await fetch(`/subscribe/${email}`, { method: 'POST' }).then(res => res.json());

    if (success) {
      return this.setState({
        success: 'All set! Thanks for signing up!',
        error: null,
        isLoading: false
      });
    }

    return this.setState({
      success: null,
      error: message,
      isLoading: false
    });
  };

  render() {
    const { success, error, isLoading } = this.state;

    return (
      <div className="subscription">
        <form className="subscription-form" onSubmit={this.subscribeUser}>
          <div className="subscription-wrapper">
            <div className="email-container">
              <input
                id="email-address"
                className={cx({ error: !!error })}
                type="email"
                name="email"
                placeholder="Your email"
                onChange={({ target: { value: email } }) => this.setState({ email })}
              />
              {error && <p>{error}</p>}
            </div>
            <button className="subscribe-button" disabled={isLoading}>
              Subscribe
            </button>
          </div>
          <div className={cx('subscription-wrapper', 'success', { hidden: !success })}>
            <div className="success-container">
              <div className="check-wrapper">
                <SVG src="/static/svg/check.svg" />
              </div>
              <p>{success}</p>
            </div>
          </div>
        </form>
        <a href="/atom" className="feed" value=" " target="blank" />
      </div>
    );
  }
}
