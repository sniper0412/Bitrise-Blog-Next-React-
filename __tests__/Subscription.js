import renderer from 'react-test-renderer';

import Subscription from '../components/Subscription';

describe('Subscription component', () => {
  console.warn('TODO: Check error and success messages');

  it('renders correctly', () => {
    const tree = renderer.create(<Subscription />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
