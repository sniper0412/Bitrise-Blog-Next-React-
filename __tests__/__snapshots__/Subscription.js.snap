// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription component renders correctly 1`] = `
<div
  className="subscription"
>
  <div
    className="subscription-form"
  >
    <div
      className="subscription-wrapper"
    >
      <div
        className="email-container"
      >
        <input
          className=""
          id="email-address"
          name="email"
          onChange={[Function]}
          placeholder="Your email"
          type="email"
        />
      </div>
      <button
        className="subscribe-button"
        onClick={[Function]}
      >
        Subscribe
      </button>
    </div>
    <div
      className="subscription-wrapper success hidden"
    >
      <div
        className="success-container"
      >
        <div
          className="check-wrapper"
        >
          <span
            className="isvg loading"
          />
        </div>
        <p />
      </div>
    </div>
  </div>
  <a
    className="feed"
    href="/atom"
    target="blank"
    value=" "
  />
</div>
`;
